### PR TITLE
Added ability to cancel infiniteDash with attack

### PR DESCRIPTION
**Pull request by JXVD**

### CODE REVIEW

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs [hunk 1]
```diff
@@ -4,25 +4,33 @@
...
 
-    public override bool AttackAllowed { get { return true; } }
+    public override bool AttackAllowed { get { return attackAllowed; } }
 
     public override MovementState DecideNextState(Vector3 velocity, Vector3 externalForces)
     {
         specialTimeLeft -= Time.deltaTime;
         //Buffer for attacks and movement specials done too early
+
+        if (specialTimeLeft < (specialTime * attackBufferWindow) && player.InputPlayer.GetButtonDown("Attack"))
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: This shouldn't be listening to the Inputs

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs [hunk 2]
```diff
@@ -36,8 +44,9 @@ public override MovementState DecideNextState(Vector3 velocity, Vector3 external
         if (isDisabled || timingState == TimingState.DONE)
         {
             hitDashEarly = false;
-            if (earlyAttackInput)
+            if (attackAllowed)
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: This probably shouldn't be happening like this. AttackAllowed isn't supposed to be treated like transition logic internally.

**Review by Juskelis — CHANGES_REQUESTED** [review]

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs (hunk 2 above)
Comment (JXVD) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: would you just want earlyAttackInput put back in as a variable?

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs (hunk 1 above)
Comment (JXVD) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: why? It needs to check for an attack input at some point and having a nested if statement felt unnecessary.

Comment (Juskelis) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: The logic for checking input already happens within Player. The design intention for AttackAllowed was to have it behave independent of user input.

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs (hunk 2 above)
Comment (Juskelis) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: That would be better; but I'm not sure we event need this logic anymore. From what I've seen, the Player's attack logic doesn't short circuit on state anymore.

Comment (JXVD) [reply Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: My understanding was that we needed to put the player into the InAttack movement state as Player.cs doesn't do that.

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs [hunk 3]
```diff
@@ -4,25 +4,43 @@
 public class OnInfiniteDash : OnDash
 {
     private bool hitDashEarly;
+    private bool attackAllowed;
+    private bool attackButtonHit;
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: Why do you have this? You're not using it anywhere.

File: Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs [hunk 4]
```diff
@@ -4,25 +4,43 @@
...
     public override MovementState DecideNextState(Vector3 velocity, Vector3 externalForces)
     {
         specialTimeLeft -= Time.deltaTime;
         //Buffer for attacks and movement specials done too early
-        if (specialTimeLeft < (specialTime * nextDashBufferWindow))
+
+        if (specialTimeLeft < (specialTime * attackBufferWindow))
         {
+            attackAllowed = true;
             if (player.InputPlayer.GetButtonDown("Attack"))
```
Comment (Juskelis) [inline Unity/Champloo/Assets/Scripts/Player/Movement States/Movement Specials/OnInfiniteDash.cs]: Get rid of this if statement, you don't need it

**Review by Juskelis — CHANGES_REQUESTED** [review]

**Review by Juskelis — APPROVED** [review]

**Review by Juskelis — APPROVED** [review]

**Review by Juskelis — APPROVED** [review]

**Pull request merged**